src/components/researchPortal/ResearchExpressionPlot.vue

```diff
@@ -208,35 +207,31 @@ export default Vue.component("ResearchExpressionPlot", {
                 entry["Max TPM"] = parseFloat(entry.maxTpm);
                 entry["nSamples"] = parseInt(entry.nSamples);
             });
-            let flatLinear = [];
-            let flatLog = [];
+            let flatBoth = [];
             for (let item of processedData) {
                 for (let tpmVal of item.tpmForAllSamples) {
-                    let flatLinearEntry = {};
-                    flatLinearEntry[this.keyAttribute] =
+                    let flatEntry = {};
+                    flatEntry[this.keyAttribute] =
                         item[this.keyAttribute];
-                    flatLinearEntry["tpm"] = tpmVal;
-                    flatLinearEntry["noise"] = Math.random();
-                    flatLinear.push(flatLinearEntry);
-
-                    let flatLogEntry = {};
-                    flatLogEntry[this.keyAttribute] = item[this.keyAttribute];
-                    flatLogEntry["tpm"] = Math.log10(tpmVal + 1);
-                    flatLogEntry["noise"] = Math.random();
-                    flatLog.push(flatLogEntry);
+                    flatEntry["linear"] = tpmVal;
+                    flatEntry["log"] = Math.log10(tpmVal + 1);
+                    flatEntry["noise"] = Math.random();
+                    flatEntry["biosample"] = Formatters.tissueFormatter(item.biosample);
+                    flatEntry["dataset"] = item.dataset;
+                    flatBoth.push(flatEntry);
                 }
             }
             this.processedData = processedData;
-            this.flatLinear = flatLinear;
-            this.flatLog = flatLog;
+            this.flatBoth = flatBoth;
             this.mapColors();
         },
         displayResults() {
             let keyAttribute = this.keyAttribute;
             let colorMap = this.colorMap;
 
-            let flatData =
-                this.logScale == "yes" ? this.flatLog : this.flatLinear;
+            let flatData = this.flatBoth;
+
+            let tpmField = this.logScale == "yes" ? "log" : "linear";
```

Comment:
Setting **logScale: false** (boolean) would be better than checking for the string **yes**, then this will be
let tpmField = this.logScale? "log" : "linear";

Generally, if a state needs to be on or off, boolean would be best and easier to work with.

Reply:
I am pretty sure I tried this and booleans interacted badly with a v-model tied to a select element. I can look at this again though.

Reply:
Never mind, this is easy. You just have to bind the boolean values. Fixed.